File: app/src/main/java/com/app/customers/domain/getQudiniInfo/GetCustomers.java
package com.app.customers.domain.getQudiniInfo;

import android.content.Context;
import android.util.Log;

import com.app.customers.data.model.QudiniCustomersToday;
import com.app.customers.data.repository.CustomersRepository;
import com.app.customers.domain.model.Customer;
import com.app.customers.executor.PostExecutionThread;
import com.app.customers.executor.ThreadExecutor;
import com.app.customers.executor.UseCaseObservable;

import java.util.List;

import io.reactivex.Observable;

/**
 * Created by Stefan
 */

public class GetCustomers extends UseCaseObservable<List<Customer>, Void>{

    private Context context;

    private CustomersRepository customersRepository;

    public GetCustomers(ThreadExecutor threadExecutor,
                        PostExecutionThread postExecutionThread,
                        Context context,
                        CustomersRepository customersRepository) {

        super(threadExecutor, postExecutionThread);

        this.context = context;

        this.customersRepository = customersRepository;

    }

    @Override
    public Observable<List<Customer>> buildUseCaseObservable(Void aVoid) {

        List<QudiniCustomersToday> customers = customersRepository.getForecastsCollection();

        Log.e("customers", "Got customers");

        return Observable.just(new ModelTransformer(customers).getCustomers());

    }

}

File: app/src/main/java/com/app/customers/domain/getQudiniInfo/GetCustomers.kt
package com.app.customers.domain.getQudiniInfo

import android.content.Context

import com.app.customers.data.model.CustomersList
import com.app.customers.data.repository.CustomersRepository
import com.app.customers.domain.model.Customer
import com.app.customers.executor.PostExecutionThread
import com.app.customers.executor.ThreadExecutor
import com.app.customers.executor.UseCaseObservable

import io.reactivex.Observable

/**
 * Created by Stefan
 */

class GetCustomers(threadExecutor: ThreadExecutor,
                   postExecutionThread: PostExecutionThread,
                   private val context: Context?,
                   private val customersRepository: CustomersRepository)
    : UseCaseObservable<List<Customer>, Void?>(threadExecutor, postExecutionThread) {

    override fun buildUseCaseObservable(params: Void?): Observable<List<Customer>> {

        val qudiniCustomers = customersRepository.customersList

        return Observable.just(ModelTransformer(qudiniCustomers
                .queueData!!
                .queue!!
                .customersToday!!)
                .getCustomers())

    }

}
